### PR TITLE
WIP: Knative 3107

### PR DESCRIPTION
**Pull request by pierDipi**
WIP mtbroker cloudevent v2

### CODE REVIEW

File: pkg/mtbroker/filter/filter_handler.go [hunk 1]
```diff
@@ -80,93 +84,132 @@ func (e sendError) Unwrap() error {
...
+	start := time.Now()
+	resp, err := h.Sender.Send(req)
+	dispatchTime := time.Since(start)
+	if err != nil {
+		h.Logger.Error("failed to dispatch message", zap.Error(err))
+		_ = h.Reporter.ReportEventDispatchTime(reportArgs, http.StatusInternalServerError, dispatchTime)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	// TODO response might contain another event
```
Comment (slinkydeveloper) [inline pkg/mtbroker/filter/filter_handler.go:204]: so you want to go from response to event right?

```
responseMessage := http.NewMessageFromHttpResponse(resp)
responseEvent, err := binding.ToEvent(responseMessage)
```

Comment (pierDipi) [reply pkg/mtbroker/filter/filter_handler.go:204]: and then where should this event be sent?

Comment (slinkydeveloper) [reply pkg/mtbroker/filter/filter_handler.go:204]: ah, i have no clue :smile: where the event was sent previously?

Comment (slinkydeveloper) [reply pkg/mtbroker/filter/filter_handler.go:204]: ah ok i see it was sent back as response in this handler, so what you should do to fill the response writer with the event is:

```
http.WriteResponseWriter(context, responseWriter, statusCode, responseMessage)
```

Comment (pierDipi) [reply pkg/mtbroker/filter/filter_handler.go:204]: I don't know :confused: 
The SDK uses reflection in that part and it's not easy to understand what it's doing.

Comment (pierDipi) [reply pkg/mtbroker/filter/filter_handler.go:204]: Cool! thank you! :+1:

**Pull request closed without merging**